### DISCUSSION TITLE
How to push/commit when you have downloaded the master?

### DISCUSSION THREAD

**Discussion by Rosie000** [community/community#21816] [New to GitHub]
<p>So I’m pretty new.</p>
<p>I have some group work on github. I was invited to the repository. I clicked to download the zip of the repository. I then did some of the code. Saved the file to my computer.</p>
<p>And now I need to push/commit that back to the master? How do I do that? I downloaded git desktop but it looks like I messed up becasue that tells me to clone, do I need to redo my work? I’m just confused.</p>
<p>Thanks in advance</p>

**Comment by logankilpatrick** [comment, marked as answer]
<p>Hey <a class="mention" href="/rosie000">@rosie000</a>, usually when you download as a zip, it’s just a full copy of the code but it’s not connected to the main repo itself. The proper way to collaborate is to git clone the repo and then make your changes to it. With that said, a potential workaround that I have had to use myself a few time is as follows: </p>
<ol>
<li>
<p>Git Clone the repo’s master branch to a known location on your computer (like the Desktop).</p>
</li>
<li>
<p>Open up file explorer or finder and then move the “.git” folder from your cloned version into the top level of the version you downloaded via zip. This should then automatically make git see the changes. </p>
</li>
<li>
<p>From there, you can use the command line (by navigating into that folder) or via GitHub desktop to commit and then push the changes. </p>
</li>
</ol>
<p>Let me know if you have any additional questions! </p>

**Comment by rambabukari-63** [comment]
how to push the code another git

**Comment by hoaithong030996-ship-it** [comment]
> So I’m pretty new.
> 
> I have some group work on github. I was invited to the repository. I clicked to download the zip of the repository. I then did some of the code. Saved the file to my computer.
> 
> And now I need to push/commit that back to the master? How do I do that? I downloaded git desktop but it looks like I messed up becasue that tells me to clone, do I need to redo my work? I’m just confused.
> 
> Thanks in advance
